v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/user/ManageUserPresenter.java

```diff
@@ -22,52 +22,99 @@
       AptoideAccountManager accountManager, ThrowableToStringMapper errorMapper,
-      FragmentNavigator fragmentNavigator) {
+      FragmentNavigator fragmentNavigator, ManageUserFragment.ViewModel userData,
+      boolean isEditProfile) {
     this.view = view;
     this.crashReport = crashReport;
     this.accountManager = accountManager;
     this.errorMapper = errorMapper;
     this.fragmentNavigator = fragmentNavigator;
+    this.userData = userData;
+    this.isEditProfile = isEditProfile;
   }
 
   @Override public void present() {
+    handleSaveDataClick();
+    handleCancelClick();
+    handleSelectImageClick();
+    onViewCreatedLoadUserData();
+  }
 
-    Observable<Void> handleSaveDataClick = view.saveUserDataButtonClick()
-        .doOnNext(__ -> view.showProgressDialog())
-        .flatMap(userData -> saveUSerData(userData).observeOn(AndroidSchedulers.mainThread())
-            .doOnCompleted(() -> view.dismissProgressDialog())
-            .doOnCompleted(() -> sendAnalytics(userData.hasImage()))
-            .doOnCompleted(() -> navigateAway(userData.isEditProfile()))
-            .onErrorResumeNext(err -> handleSaveUserDataError(err, userData.isEditProfile()))
-            .toObservable())
-        .retry()
-        .map(__ -> null);
+  @Override public void saveState(Bundle state) {
+    // does nothing
+  }
 
-    Observable<Void> handleCancelClick = view.cancelButtonClick()
-        .doOnNext(__ -> navigateBack());
+  @Override public void restoreState(Bundle state) {
+    // does nothing
+  }
 
-    Observable<Void> handleSelectImageClick = view.selectUserImageClick()
-        .retry()
-        .doOnNext(__ -> view.showLoadImageDialog());
+  private void onViewCreatedLoadUserData() {
+    view.getLifecycle()
+        .filter(event -> event == View.LifecycleEvent.CREATE)
+        .flatMapSingle(__ -> accountManager.accountStatus()
+            .first()
+            .toSingle())
+        .map(userAccount -> {
+          if (userData == null && isEditProfile) {
+            return new ManageUserFragment.ViewModel(userAccount.getNickname(), null,
+                userAccount.getAvatar());
+          } else {
+            return userData;
+          }
+        })
+        .filter(data -> data != null)
+        .observeOn(AndroidSchedulers.mainThread())
+        .subscribe(data -> {
+          view.setUserImage(data.getImagePathToView());
+          view.setUserName(data.getName());
+        }, err -> crashReport.log(err));
+  }
 
+  private void handleSaveDataClick() {
     view.getLifecycle()
         .filter(event -> event == View.LifecycleEvent.CREATE)
-        .flatMap(
-            __ -> Observable.merge(handleCancelClick, handleSaveDataClick, handleSelectImageClick))
+        .flatMap(__ -> view.saveUserDataButtonClick()
+            .doOnNext(__2 -> view.showProgressDialog())
+            .flatMap(userData -> saveUserData(userData))
+            .retry())
```

Comment:
on save user data, some error could happen and we need to re-subscribe to the click of UI button that triggers the saving of the data. otherwise in an error with a first click, the button would do nothing on a second click.